clippy_lints/src/unconditional_recursion.rs

```diff
@@ -285,15 +286,13 @@ where
     'tcx: 'a,
 {
     type NestedFilter = nested_filter::OnlyBodies;
+    type Result = ControlFlow<()>;
 
     fn nested_visit_map(&mut self) -> Self::Map {
         self.map
     }
 
-    fn visit_expr(&mut self, expr: &'tcx Expr<'tcx>) {
-        if self.found_default_call {
-            return;
-        }
+    fn visit_expr(&mut self, expr: &'tcx Expr<'tcx>) -> ControlFlow<()> {
         walk_expr(self, expr);
```

Comment:
```suggestion
        walk_expr(self, expr)?;
```
propagating any `Break`s

Reply:
pls also double check that the `Break`s from the other `walk_*` calls are propagated too in case I missed one :3

Reply:
I'm having some problems with this suggestion. I'm having this problem with other `walk_expr`s:

```
error[E0308]: `?` operator has incompatible types
   --> clippy_lints/src/if_let_mutex.rs:100:9
    |
96  | ...n visit_expr(&mut self, expr: &'tcx Expr<'_>) -> ControlFlow<&'tcx Expr<'tcx>...
    |                                                     ----------------------------- expected `ControlFlow<&'tcx rustc_hir::Expr<'tcx>>` because of return type
...
100 | ...   visit::walk_expr(self, expr)?
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^-
    |       |                           |
    |       |                           help: try removing this `?`
    |       expected `ControlFlow<&Expr<'_>>`, found `()`
    |
    = note: `?` operator cannot convert from `()` to `ControlFlow<&'tcx rustc_hir::Expr<'tcx>>`
    = note:   expected enum `ControlFlow<&'tcx rustc_hir::Expr<'tcx>>`
            found unit type `()`
```

Reply:
Ah, my second comment might've been confusing. It was unrelated to the `?` (just meant that we should make sure all `ControlFlow::Break`s are properly handled).
Looks like the `?` is only needed in `unconditional_recursion` specifically, since it would otherwise be ignored.

In the `if_let_mutex` case (and all other cases where `walk_expr` is the last expression in the function) just `walk_expr(self, expr)`  without the `?` should be enough since its immediately returned.